osbs/tekton.py

```diff
@@ -697,11 +697,8 @@ def wait_for_taskruns(self):
                 continue
             for task_run_name, task_run_data in task_runs.items():
                 if task_run_name not in watched_task_runs:
-                    watched_task_runs.append(task_run_name)
+                    watched_task_runs.add(task_run_name)
                     yield task_run_data['pipelineTaskName'], task_run_name
-            # all task runs accounted for
-            if len(pipeline_run['status']['pipelineSpec']['tasks']) == len(task_runs):
-                return
```

Comment:
Now that I think about it, this commit on its own only makes the problem worse, by removing the only early-return condition that this method had. It's fixed in the next one, but maybe it should be a single commit.